### ISSUE TITLE
Output template option output 'NA' for playlist option

### ISSUE THREAD

**Issue by athiosh** [kieraneglin/pinchflat#171]
I am new to this and trying to setup a profile for playlist but all of the playlist option below would only return 'NA'. Is there a different way to output the playlist name, playlist id, and playlist index?

/{{ playlist }} [{{ playlist_id}}]/{{ playlist_index }} - {{ title }} [{{ id }}].{{ ext }}

**Comment by kieraneglin** [comment]
Hey there, thanks for the report!

Due to the way that Pinchflat breaks indexing and downloading into separate steps, certain playlist-related data cannot be used in output templates. That said, I've just added in some new output template options in #172 that map `playlist_id` to `{{ source_collection_id }}` and `playlist_title` to `{{ source_collection_name }}`. #172 will be out in the next release which should go out later today or sometime tomorrow.

`playlist_index` is an annoying one - I _really_ want it to work, but in my testing it's so inconsistent as to be unusable in its current state ([comment experiencing a similar issue](https://github.com/yt-dlp/yt-dlp/pull/302#issuecomment-1925348547)). 

To recap:

- `playlist_id` -> `source_collection_id`
- `playlist_title` -> `source_collection_name`
- `playlist` -> either `source_collection_id` or `source_collection_name` (I wouldn't recommend the `playlist` attribute anyway - use one of the above attributes instead)
- `playlist_index` -> not currently possible and not planned until it works more consistently

---

Please let me know if you have any questions!

**Comment by eleetsrelyt** [comment]
This is my output path template: 
/shows/{{ source_custom_name }}/{{source_collection_name}} [{{source_collection_id}}]/{{ title }} - [{{id}}].{{ ext }}

For the source URL: `https://www.youtube.com/@TetraLogical/playlists` I get this file structure:

> shows/tetra test 1/Quick accessibility tests (British Sign Language version) [PLTqm2yVMMUKVJsYABedyzqzoYDhxaFA2S]/Browsing with a desktop screen reader (British Sign Language version) - [gcNWPYnWtSc].mp4

For the source URL: `https://www.youtube.com/@TetraLogical` I get this file structure:

> shows/tetra test 2/TetraLogical [UC8u1Gauw4HH-dM7j-IwCT7Q]/Browsing with a desktop screen reader (British Sign Language version) - [gcNWPYnWtSc].mp4

Both options give only 1 subfolder with all the videos. Is there any way to get every playlist put into its own folder with youtube's PL* attribute?

![Screenshot 2024-04-10 at 3 43 20 PM](https://github.com/kieraneglin/pinchflat/assets/4566433/3dd0ddbb-989d-4dae-96af-e734ad7b820d)

**Comment by kieraneglin** [comment]
@eleetsrelyt I'm not 100% sure I follow but I'll try my best!

I think you have a few options here:

- This one is better but only works if _every_ video is from the same channel:

```
 /shows/{{ channel }}/{{source_collection_name}} [{{source_collection_id}}]/{{ title }} - [{{id}}].{{ ext }}
 ```
 
 - You can also use the same `Custom Name` for each source and use the output template that you already have. Not ideal for organization, though
 - You can almost certainly use some `yt-dlp` [output template magic](https://github.com/yt-dlp/yt-dlp?tab=readme-ov-file#output-template) to generate a path that would work
 
 ---
 
 If that doesn't answer your question, would you mind posting an example of your desired output paths? That would help me determine the best approach

**Comment by eleetsrelyt** [comment]
I've been doing the 2nd option as you suggest. It works, but it's messy. 

For some reason, the output path you suggest still dumps all the playlists and videos into the same subfolder under the `channel` folder, and the source collection name and id change depending on what link I use for the source URL (UC* vs PL*). Ideally, I want something that [YouTube-Agent bundle](https://github.com/ZeroQI/YouTube-Agent.bundle) for Plex recognizes, and I want the seasons to be playlists rather than dates:

```
CaRtOoNz [UCdQWs2nw6w77Rw0t-37a4OA]/
- Ben and Ed [PLTqm2yVMMUKVJsYABedyzqzoYDhxaFA2S]/
  - Ben and Ed _ 'My Zombie Best Friend!' (I Didn't Need Those Legs Anyway!) [fRFr7L_qgEo].mkv
  - Ben and Ed _ 'Clownin Around!' (F_ck You Neck-Beard!) [Nh9eILgD5N4].mkv
- Golf With Your Friends [PLTqm2yVMMUKVxQ6eYth4_vvbM3IcrSPlk]/
  - Golf With Friends _ HOLE IN ONE...THOUSAND! (w_ H2O Delirious, Bryce, & Ohmwrecker) [81er8CP24h8].mkv
  - Golf With Friends _ GOLF LIKE AN EGYPTIAN! (w_ H2O Delirious, Bryce, & Ohmwrecker) [gKYid-SjDiE].mkv

H2ODelirious [UCClNRixXlagwAd--5MwJKCw]/
- Ben and Ed [PLTqm2yVMMUKWpUN1zLyxNchyomZXOoaXJ]/
  - Ben And Ed Ep.1 (MUST SAVE BEN) BRAINNNNNSSSS [9YeXl28l9Yg].mkv
  - Ben And Ed - Blood Party - ANGRYLIRIOUS!!!!! (I CAN DO THIS!) [BEDE2z3G3hY].mkv
- Golf With Your Friends [PLTqm2yVMMUKWTr9XWdW5hJ9tk512Ow0SE]/
  - Golf With Your Friends - 1st Time Playing! 'Professionals' [wxS52xI_W_Y].mkv
  - Golf With Your Friends - Hitting Balls, Stroking Out! [GdLon0CCEXE].mkv
```

**Comment by kieraneglin** [comment]
Got it! Thanks for the update

I don't have a good solution for you at the moment, but I think I could add the ability to optionally override the output template on a per-source basis which would solve this issue while keeping things organized.

I'll think about the implications of that and start working on it tomorrow evening or so! I'm tracking its progress in #178

**Comment by kieraneglin** [comment]
This _should_ be resolved by #179 which will be released as part of `v0.1.13` later today or sometime tomorrow.  It's still a little verbose, but it's much better than before imo.

Now, you can go to your source and set it's output template override to something like `/CaRtOoNz [UCdQWs2nw6w77Rw0t-37a4OA]/{{source_collection_name}} [{{source_collection_id}}]/{{ title }} - [{{id}}].{{ ext }}` which would namespace that source under the parent `CaRtOoNz [UCdQWs2nw6w77Rw0t-37a4OA]/` directory

**Comment by eleetsrelyt** [comment]
For some reason, this still doesn't give me the desired folder structure when I use an entire channel as the input source. I'll just put the playlists in one at a time to get the output I'm looking for. Thanks!

**Comment by ckambler** [comment]
I'm still having this issue, specifically with the "playlist_index" field. I'm finding several playlists where sometimes videos are uploaded out of order, so the uploaded date is not a useful order for the playlist, and the order is set manually by the channel. If i use playlist_index as a filename field, it's just coming through as "NA". Are there any ways to get playlist_index working so that the files are named in order of the playlist order? The s##e## naming scheme is the only way i know to get Plex to recognize these properly, as you can't re-arrange episode order in Plex after the fact.
